pulpcore/app/models/content.py

```diff
@@ -252,6 +252,19 @@ class Meta:
         verbose_name_plural = 'content'
         unique_together = ()
 
+    # This decorator should go away when (if ever) it is fixed in Django
+    # https://code.djangoproject.com/ticket/30041
+    @classmethod
+    def __init_subclass__(cls, **kwargs):
+        from django.contrib.contenttypes.models import ContentType
+        content_type = ContentType.objects.get_for_model(cls)
+        app_label = content_type.app_label
```

Comment:
These three lines is a way I was able to find how to get app_label.
This PR will work if you have already installed Pulp.
If you try to create migrations from scratch, it will fail.
```
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/pulp/bin/pulp-manager", line 11, in <module>
    load_entry_point('pulpcore', 'console_scripts', 'pulp-manager')()
  File "/home/vagrant/devel/pulp/pulpcore/app/entry_points.py", line 11, in pulp_manager_entry_point
    execute_from_command_line(sys.argv)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/core/management/__init__.py", line 357, in execute
    django.setup()
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/apps/registry.py", line 112, in populate
    app_config.import_models()
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/apps/config.py", line 198, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/vagrant/devel/pulp_file/pulp_file/app/models.py", line 11, in <module>
    class FileContent(Content):
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/models/base.py", line 78, in __new__
    new_class = super_new(cls, name, bases, new_attrs, **kwargs)
  File "/home/vagrant/devel/pulp/pulpcore/app/models/content.py", line 261, in __init_subclass__
    content_type = ContentType.objects.get_for_model(cls)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/contrib/contenttypes/models.py", line 51, in get_for_model
    ct = self.get(app_label=opts.app_label, model=opts.model_name)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/models/query.py", line 393, in get
    num = len(clone)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/models/query.py", line 250, in __len__
    self._fetch_all()
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/models/query.py", line 1186, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/models/query.py", line 54, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/models/sql/compiler.py", line 1065, in execute_sql
    cursor.execute(sql, params)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/vagrant/.virtualenvs/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: relation "django_content_type" does not exist
LINE 1: ..."."app_label", "django_content_type"."model" FROM "django_co...
```
I think it happens due to the order of how Django apps are populated. I'm not sure if it's possible to force Django to fully install/populate some apps before models.
Any ideas on how to get app_label in a different way or how to workaround the order ^ ?